### PR TITLE
F2F-580:Added front api mapping resources 

### PR DESCRIPTION
**Pull request by AyeshaKhanGDS**
### What changed

Added custom domain, apirecord, and api mapping resources to FE repo which were previously in infra repo

### Why did it change

To remove circular dependencies

### Issue tracking

- [F2F-580](https://govukverify.atlassian.net/browse/F2F-580)

[F2F-580]: https://govukverify.atlassian.net/browse/F2F-580?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -574,67 +586,50 @@ Resources:
...
+      - IsNotDevelopment
+      - !Sub
+        - "www.${DNSSUFFIX}"
+        - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+      - !Sub
+        - "${AWS::StackName}.${DNSSUFFIX}"
+        - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+      Type: A
+      HostedZoneId: !Sub "{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}"
+    #  HostedZoneId: !ImportValue PublicHostedZoneId
```
Comment (darrendunford) [inline template.yaml:620]:
```suggestion
```

File: template.yaml [hunk 2]
```diff
@@ -343,8 +353,8 @@ Resources:
           Environment:
             - Name: API_BASE_URL
               Value: !FindInMap [EnvironmentVariables, !Ref Environment, APIBASEURL ]
-            - Name: EXTERNAL_WEBSITE_HOST
-              Value: !FindInMap [EnvironmentVariables, !Ref Environment, EXTERNALWEBSITEHOST ]
+            - Name: FRONT_END_CUSTOM_DOMAIN
```
Comment (darrendunford) [inline template.yaml:356]: Can you check with an FE engineer that this is the right env var to pass the custom domain name in to the container on?

File: template.yaml [hunk 3]
```diff
@@ -89,33 +89,43 @@ Mappings:
     dev:
       EXTERNALWEBSITEHOST: "https://review-c.dev.account.gov.uk"
       APIBASEURL: "https://api.review-c.dev.account.gov.uk"
+      DNSSUFFIX: "review-c.dev.account.gov.uk"
       SESSIONTABLENAME: "cic-front-sessions-dev"
       GTMID: "TK92W68"
       ANALYTICSDOMAIN: "dev.account.gov.uk"
+      PROXYURL: "cic-cri-outbound-proxy-proxy.review-c.dev.account.gov.uk"
```
[the remark this answers is not in the document]
Comment (AyeshaKhanGDS) [reply template.yaml:96]: Proxy references removed

File: template.yaml (hunk 2 above)
Comment (AyeshaKhanGDS) [reply template.yaml:356]: now replaced with exerternal website host

File: template.yaml (hunk 1 above)
Comment (AyeshaKhanGDS) [reply template.yaml:620]: removed

**Review by richa-misra — APPROVED** [review]

**Review by richa-misra — APPROVED** [review]

**Pull request merged**